terraform/deployments/cluster-infrastructure/external_secrets_iam.tf

```diff
@@ -0,0 +1,50 @@
+# IAM role and policy to enable the k8s external-secrets operator to talk to
+# AWS APIs to access and manage SecretsManager secrets.
+#
+# The k8s side of the autoscaler is in
+# ../cluster-services/external_secrets.tf.
+
+locals {
+  external_secrets_service_account_namespace = "secrets"
+  external_secrets_service_account_name      = "external-secrets"
+}
+
+# The module block external_secrets_iam_role is taken from
+# https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/examples/irsa/irsa.tf,
+# which is Apache-2.0 licenced:
+# https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/LICENSE
+#
+# Significant changes from upstream:
+# - The aws_iam_policy uses a constructed name instead of name_prefix.
+module "external_secrets_iam_role" {
+  source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version                       = "4.3.0"
+  create_role                   = true
+  role_name                     = local.external_secrets_service_account_name
+  provider_url                  = local.cluster_oidc_issuer
+  role_policy_arns              = [aws_iam_policy.external_secrets.arn]
+  oidc_fully_qualified_subjects = ["system:serviceaccount:${local.external_secrets_service_account_namespace}:${local.external_secrets_service_account_name}"]
+}
+
+resource "aws_iam_policy" "external_secrets" {
+  name        = "EKSExternalSecrets-${var.cluster_name}"
+  description = "EKS ${local.external_secrets_service_account_name} policy for cluster ${module.eks.cluster_id}"
+  policy      = data.aws_iam_policy_document.external_secrets.json
+}
+
+data "aws_iam_policy_document" "external_secrets" {
+  statement {
+    sid    = "externalSecretsSecretsManager"
+    effect = "Allow"
+    actions = [
+      "secretsmanager:GetResourcePolicy",
+      "secretsmanager:GetSecretValue",
+      "secretsmanager:DescribeSecret",
+      "secretsmanager:ListSecretVersionIds"
+    ]
+
+    resources = [
+      "arn:aws:secretsmanager:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:secret:*",
```

Comment:
Should we use a prefix for our secrets (`govuk/` or such), just to make sure they're separate from anything else which may end up in the account's Secrets Manager?

Reply:
Prefixing with the `${cluster_name}/` sounds sensible to me. Any downsides?

Reply:
Only that it explicitly binds a given set of secrets to a specific cluster, but that's likely no bad thing.

Reply:
...for now anyway. _Maybe_ at some stage we'll have multiple application environments in a single (non-prod) cluster, but as always we can burn this bridge when we come to it.